src/components/User/Username.tsx

```diff
@@ -17,37 +17,44 @@ import './Username.css'
 type Props = SizeProps & {
   address: string
   linked?: boolean
-  showName?: boolean
   className?: string
   blockieScale?: number
+  iconOnly?: boolean
 }
 
-const Username = ({ address, size, linked, showName = true, blockieScale = 3, className }: Props) => {
+const Username = ({ address, size, linked, iconOnly = false, blockieScale = 3, className}: Props) => {
   const { profile } = useProfile(address)
+  const userHasCustomProfile = profile && !profile.isDefaultProfile
+  const profileHasName = userHasCustomProfile && profile.name && profile.name.length > 0
 
   const userElement = (
     <>
-      {profile && !profile.isDefaultProfile && <Avatar size={size || 'mini'} address={address} />}
-      {profile && showName && profile.name}
-      {profile && !showName && <Address value={address} />}
-      {(!profile || profile.isDefaultProfile) && !!address && (
+      {userHasCustomProfile && <>
```

Comment:
What about renaming `hasDclAvatar` to `hasDclProfile` in useProfile? It is the same condition!